src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -62,62 +71,66 @@ protected CompletableFuture<Void> processHoldings(OrderLineUpdateInstanceHolder
 
 
   private CompletableFuture<Void> moveHoldings(OrderLineUpdateInstanceHolder holder, String newInstanceId, RequestContext requestContext) {
-    List<String> holdingIds = holder.getStoragePoLine()
-      .getLocations()
-      .stream()
-      .map(Location::getHoldingId)
-      .filter(Objects::nonNull)
-      .collect(toList());
-    holdingIds.forEach(id -> holder.addHoldingRefsToStoragePatchOrderLineRequest(id, id));
-
-    return inventoryManager.getHoldingsByIds(holdingIds, requestContext)
-        .thenCompose(holdings -> inventoryManager.updateInstanceForHoldingRecords(holdings, newInstanceId, requestContext));
+    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine()), requestContext)
+      // Extraction unique holdingIds from pieces and POL
```

Comment:
nitpick: think you mean "extract" not "extraction"

Reply:
I decided to extract as new method following by best practices, are you ok?

Reply:
I was only referring to your grammar in the comment. I am good with putting things in a new method.